deltachat-ffi/deltachat.h

```diff
@@ -2544,6 +2602,21 @@ dc_context_t*  dc_accounts_get_selected_account (dc_accounts_t* accounts);
 int            dc_accounts_select_account       (dc_accounts_t* accounts, uint32_t account_id);
 
 
+/**
+ * Get the lowest connectivity of all accounts managed by the account manager
```

Comment:
As far as I know, this is needed for iOS so it can get to sleep when all the accounts have stopped working and got into `CONNECTED` state. It's better to describe it here so it's clear that it is not expected to be used in the UI.

What if one of the accounts is stuck in the `NOT_CONNECTED` state? Will it prevent the iOS app from going to sleep?

Reply:
> What if one of the accounts is stuck in the `NOT_CONNECTED` state? Will it prevent the iOS app from going to sleep?

Currently, unfortunately, yes. IIRC we said that's OK as there usually won't be any account stuck, but still a bit unfortunate.

Reply:
But the iOS app can't stay awake for longer than 30s anyway, then the system will kill it, so it won't stay awake forever. (If it stays awake shorter, it will be allowed to wake up sooner.)

Reply:
Because directly after calling start_io() and maybe_network(), the state is going to be NOT_CONNECTED for a short period of time.

This could be fixed by this patch though; do you think we should do this? Because, it does further increase the complexity

```diff
diff --git a/python/tests/test_account.py b/python/tests/test_account.py
index e0302d2e..a29d2bf8 100644
--- a/python/tests/test_account.py
+++ b/python/tests/test_account.py
@@ -2055,7 +2055,6 @@ def test_connectivity(self, acfactory, lp, inbox_watch):
         ac1._evtracker.wait_for_connectivity(const.DC_CONNECTIVITY_NOT_CONNECTED)
 
         ac1.start_io()
-        ac1._evtracker.wait_for_connectivity(const.DC_CONNECTIVITY_CONNECTING)
         ac1._evtracker.wait_for_connectivity_change(const.DC_CONNECTIVITY_CONNECTING, const.DC_CONNECTIVITY_CONNECTED)
 
         lp.sec("Test that after calling start_io(), maybe_network() and waiting for connectivity `CONNECTED`, " +
@@ -2100,7 +2099,7 @@ def test_connectivity(self, acfactory, lp, inbox_watch):
         ac1.stop_io()
         ac1._evtracker.wait_for_connectivity(const.DC_CONNECTIVITY_NOT_CONNECTED)
         ac1.start_io()
-        ac1._evtracker.wait_for_connectivity(const.DC_CONNECTIVITY_CONNECTING)
+        assert ac1.get_connectivity() == const.DC_CONNECTIVITY_CONNECTING
         ac1._evtracker.wait_for_connectivity(const.DC_CONNECTIVITY_NOT_CONNECTED)
 
     def test_fetch_deleted_msg(self, acfactory, lp):
diff --git a/src/scheduler.rs b/src/scheduler.rs
index 739fe6da..9333e99b 100644
--- a/src/scheduler.rs
+++ b/src/scheduler.rs
@@ -324,6 +324,12 @@ pub async fn start(&mut self, ctx: Context) -> Result<()> {
         let mut sentbox_handle = None;
         let (smtp_start_send, smtp_start_recv) = channel::bounded(1);
 
+        // TODO explain why we do that, this is probably the trickiest part
+        inbox_handlers
+            .connection
+            .connectivity
+            .set_connecting(&ctx)
+            .await;
         let inbox_handle = {
             let ctx = ctx.clone();
             Some(task::spawn(async move {
```

Reply:
There is going to be another problem if we do this (make a function that checks directly that there are no accounts in the WORKING or CONNECTING state):

If the user has only one account (i.e. probably the standard for the time being), and one folder of this account has a problem, the whole connectivity will be NotConnected and iOS will directly go back to sleep without fetching at all.

In conclusion, our decision to use the result of `get_connectivity()` has these disadvantages:
- What I described in this message here
- what I described in the previous message
- the SMTP state is ignored, i.e. sending messages out could be aborted
- all UIs will get an extra connectivity `INTERRUPTING_IDLE` which most of them don't need

...and I think I should just make a completely separate function for this that looks at the `DetailedConnectivity` status.